web/src/client/storage.js

```diff
@@ -245,172 +247,135 @@ class DevicesManager {
    * @returns {Promise<StorageDevice[]>}
    */
   async getDevices() {
-    /** @type {(path: string, dbusDevices: object[]) => StorageDevice} */
-    const buildDevice = (path, dbusDevices) => {
-      /** @type {(device: StorageDevice, deviceProperties: object) => void} */
-      const addDeviceProperties = (device, deviceProperties) => {
-        device.sid = deviceProperties.SID.v;
-        device.name = deviceProperties.Name.v;
-        device.description = deviceProperties.Description.v;
+    const buildDevice = (jsonDevice, jsonDevices) => {
```

Comment:
I have the impression that we are abusing inline functions. AFAIS, this function does not have any closure and, moreover, it defines many other internal functions. I would add this to our list of things to discuss.